pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -1291,6 +1287,35 @@ func (shcPlaybookContext *SHCPlaybookContext) triggerBundlePush() error {
 	return nil
 }
 
+// getClusterScopedAppsLocOnPod returns the cluster apps directory
+func getClusterScopedAppsLocOnPod(cr splcommon.MetaObject) string {
+	switch cr.GetObjectKind().GroupVersionKind().Kind {
+	case "ClusterMaster":
+		return idxcAppsLocationOnClusterManager
+	case "SearchHeadCluster":
+		return shcAppsLocationOnDeployer
+	default:
+		return ""
+	}
+}
+
+// adjustClusterAppsFilePermissions sets the file permissions to +550
+func adjustClusterAppsFilePermissions(podExecClient splutil.PodExecClientImpl) error {
```

Comment:
I think it's better to create a more generic function so that anybody can use it to change file permission for any directory. So maybe something like `adjustFilePermissions(podExecClient splutil.PodExecClientImpl, dirPath string)` . So the callers of this API can set the diPath and this function would just set the file permissions for that path.

Reply:
TBH it's not about 1 command but what we are trying to achieve and the usability in the future. But I leave it up to you if you want to change it or not :)

Reply:
Agree, but here the use case is a bit narrow for this situation(recursive, append, permission value, etc..), without much visibility for future use cases. If we want to have a generic API, then need the options for all those R, +, and <value>. At this time we are not seeing much scope, leaving it this way for now. But, definitely, if we see a need, we can bring in those extra options and make it a generic API.